saleor/graphql/product/filters.py

```diff
@@ -511,32 +512,14 @@ def filter_collections(qs, _, value):
 
 
 def _filter_products_is_published(qs, _, value, channel_slug):
-    channel = Channel.objects.using(qs.db).filter(slug=channel_slug).values("pk")
-    product_channel_listings = (
-        ProductChannelListing.objects.using(qs.db)
-        .filter(Exists(channel.filter(pk=OuterRef("channel_id"))), is_published=value)
-        .values("product_id")
-    )
-
-    # Filter out product for which there is no variant with price
-    variant_channel_listings = (
-        ProductVariantChannelListing.objects.using(qs.db)
-        .filter(
-            Exists(channel.filter(pk=OuterRef("channel_id"))),
-            price_amount__isnull=False,
+    if channel := Channel.objects.using(qs.db).filter(slug=channel_slug).first():
```

Comment:
Are you sure that your solution will behave in the same way for not existing slug?